src/util/run.cpp

```diff
@@ -77,29 +112,9 @@ int run(
   return status;
 
   #else
-  int stdin_fd=STDIN_FILENO;
-
-  if(!std_input.empty())
-  {
-    stdin_fd=open(std_input.c_str(), O_RDONLY);
-    if(stdin_fd==-1)
-    {
-      perror("Failed to open stdin copy");
-      return 1;
-    }
-  }
-
-  int stdout_fd=STDOUT_FILENO;
-
-  if(!std_output.empty())
-  {
-    stdout_fd=open(std_output.c_str(), O_CREAT|O_WRONLY, S_IRUSR|S_IWUSR);
-    if(stdout_fd==-1)
-    {
-      perror("Failed to open stdout copy");
-      return 1;
-    }
-  }
+  int stdin_fd = stdio_redirection(STDIN_FILENO, std_input);
+  int stdout_fd = stdio_redirection(STDOUT_FILENO, std_input);
+  int stderr_fd = stdio_redirection(STDERR_FILENO, std_error);
```

Comment:
This needs error-checking:
`if(stdin_fd == -1 || stdout_fd == -1 || stderr_fd == -1) return 1;`

Reply:
done